### PR TITLE
Add re-caching tests when nodes are added/deleted.

### PR DESCRIPTION
**Pull request by lekhajee**
Needs https://github.com/rackerlabs/mimic/pull/387 to be merged and a Mimic release.

### CODE REVIEW

File: arsenal/tests/functional/base.py [hunk 1]
```diff
@@ -224,12 +231,27 @@ def get_cached_ironic_nodes_by_flavor(self):
...
+
+    def calculate_percentage_to_be_cached(self, total_nodes, percentage,
+                                          by_flavor=True):
         """Calulates the nodes to be cached given the percentage and the
         total_nodes.
         """
-        return (math.floor((total_nodes / self.flavors) *
-                           percentage) * self.flavors)
+        if by_flavor:
+            return (math.floor((total_nodes / self.flavors) *
```
Comment (ClifHouck) [inline arsenal/tests/functional/base.py]: Another thought here: are the number of nodes in each flavor always going to be the same in these test?

File: arsenal/tests/functional/base.py [hunk 2]
```diff
@@ -259,3 +281,32 @@ def list_ironic_nodes_by_image(self, node_list, count=False):
                 nodes_per_image_count[key] = len(value)
             return nodes_per_image_count
         return nodes_per_image
+
+    def add_new_nodes_to_mimic(self, num=1, memory_mb=131072):
+        """Add the `num` number of nodes in mimic.
+        By default adds onmetal-io1 flavors.
+        """
+        request_json = json.dumps({'properties': {'memory_mb': 131072}})
```
Comment (ClifHouck) [inline arsenal/tests/functional/base.py]: I think 131072 is supposed to be memory_mb here since you've got an argument for it.

**Comment by ClifHouck** [comment]
+1 LGTM except for one problem, and one question on calculating cached nodes per flavor

will wait to merge until tests pass

File: arsenal/tests/functional/base.py (hunk 1 above)
Comment (lekhajee) [inline arsenal/tests/functional/base.py]: Yes, currently mimic is configured to have equal amounts of nodes for each image. A control plane would have to be introduced for the glance admin api to make that configurable - https://github.com/rackerlabs/mimic/issues/360.

**Pull request merged**